crates/nu-protocol/src/errors/config_error.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{ShellError, Span, Type};
+use miette::Diagnostic;
+use thiserror::Error;
+
+/// The errors that may occur when updating the config
+#[derive(Clone, Debug, PartialEq, Error, Diagnostic)]
+pub enum ConfigError {
+    #[error("Type mismatch at {path}")]
+    #[diagnostic(code(nu::shell::type_mismatch))]
+    TypeMismatch {
+        path: String,
+        expected: Type,
+        actual: Type,
+        #[label = "expected {expected}, but got {actual}"]
+        span: Span,
+    },
+    #[error("Invalid value for {path}")]
+    #[diagnostic(code(nu::shell::invalid_value))]
+    InvalidValue {
+        path: String,
+        valid: String,
+        actual: String,
+        #[label = "expected {valid}, but got {actual}"]
+        span: Span,
+    },
+    #[error("Unknown config option: {path}")]
+    #[diagnostic(code(nu::shell::unknown_config_option))]
+    UnknownOption {
+        path: String,
+        #[label("remove this")]
+        span: Span,
+    },
+    #[error("{path} requires a '{column}' column")]
+    #[diagnostic(code(nu::shell::missing_required_column))]
+    MissingRequiredColumn {
+        path: String,
+        column: &'static str,
+        #[label("has no '{column}' column")]
+        span: Span,
+    },
+    #[error(transparent)]
+    #[diagnostic(transparent)]
+    ShellError(#[from] ShellError),
```

Comment:
Mhh doesn't this kind of undermine the value of having this separate enum if we can also encounter generic crap?

Reply:
I intended for this to be temporary (it was for the deprecated error on `use_grid_icons`).

Reply:
Maybe worth a deprecation variant instead? Likely that we have some deprecated keys or values from time to time.

But, that could also be something to refine.

Reply:
Yeah, good suggestion. I added a case for deprecation. Although, I still wasn't able to remove the `Shellerror` case, since `Vec::from_value` returns a `Result`. But this can be removed later if we more strongly type that part of the config.